src/test/java/com/teragrep/pth10/steps/teragrep/bloomfilter/FilterTypesTest.java

```diff
@@ -49,92 +49,104 @@
 import com.typesafe.config.ConfigFactory;
 import org.apache.spark.util.sketch.BloomFilter;
 import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
 
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 import java.util.Map;
 import java.util.Properties;
 
 import static org.junit.jupiter.api.Assertions.*;
 
 class FilterTypesTest {
 
+    Connection conn;
+    final String username = "sa";
+    final String password = "";
+    final String connectionUrl = "jdbc:h2:mem:test;MODE=MariaDB;DATABASE_TO_LOWER=TRUE;CASE_INSENSITIVE_IDENTIFIERS=TRUE";
+
+    @BeforeEach
+    public void setup() {
+        Config config = ConfigFactory.parseProperties(defaultProperties());
+        this.conn = new LazyConnection(config).get();
+        Assertions.assertDoesNotThrow(() -> {
+            conn.prepareStatement("DROP ALL OBJECTS").execute(); // h2 clear database
+        });
+        Assertions.assertDoesNotThrow(() -> {
+            Class.forName("org.h2.Driver");
+        });
+        String createFilterType = "CREATE TABLE `filtertype` ("
+                + "`id`               bigint(20) UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,"
+                + "`expectedElements` bigint(20) NOT NULL," + "`targetFpp`        DOUBLE UNSIGNED NOT NULL,"
+                + "`pattern`          VARCHAR(255) NOT NULL)";
+        Assertions.assertDoesNotThrow(() -> {
+            conn.prepareStatement(createFilterType).execute();
+        });
+
+    }
     @Test
     public void testSortedMapMethod() {
-        Properties properties = new Properties();
-        properties
-                .put(
-                        "dpl.pth_06.bloom.db.fields",
-                        "[" + "{expected: 1000, fpp: 0.01}," + "{expected: 2000, fpp: 0.01},"
-                                + "{expected: 3000, fpp: 0.01}" + "]"
-                );
-        Config config = ConfigFactory.parseProperties(properties);
+        Config config = ConfigFactory.parseProperties(defaultProperties());
         FilterTypes filterTypes = new FilterTypes(config);
         Map<Long, Double> resultMap = filterTypes.sortedMap();
         assertEquals(0.01, resultMap.get(1000L));
-        assertEquals(0.01, resultMap.get(2000L));
-        assertEquals(0.01, resultMap.get(3000L));
+        assertEquals(0.02, resultMap.get(2000L));
```

Comment:
Was about to comment about the changed assertions but it looks fine, default properties are a bit different, but it seems to be an improvement on the tests.

Reply:
This change better ensures that the FPP values are also applied correctly